### PR TITLE
[agent] AL-12097: create or update transform file on agent apply

### PR DESCRIPTION
**Pull request by AlexMuliar**

### CODE REVIEW

File: agent/src/agent/cli/apply.py [hunk 1]
```diff
@@ -51,6 +51,9 @@ def populate_pipeline_from_file(file):
         try:
             if 'pipeline_id' not in config:
                 raise Exception(f'Pipeline configs must contain a `pipeline_id` field, check {file.name}')
+            if config.get('transform'):
+                with open(config['transform']['file'], 'w+') as f:
+                    f.write(config['transform']['config'])
```
Comment (vpoliukh) [inline agent/src/agent/cli/apply.py:56]: Need to check that 'file' key in config['transform'], will be an exception if transform declared without file.
Sees snmp_table pipeline for example

**Review by vpoliukh — CHANGES_REQUESTED** [review]

**Comment by dariakharlan** [comment]
In the ticket I wrote ```export csv config with pipelines to a file (configs like transform files etc)``` it's not implemented here. I meant to export the transform file (for some pipelines it can be also a query file) with config when we do `agent pipeline export`. Let me know if I need to explain it more

**Review by dariakharlan — CHANGES_REQUESTED** [review]
In the ticket I wrote ```export csv config with pipelines to a file (configs like transform files etc)```, it means exporting the transform file (for some pipelines it can be also a query file) with config when we do `agent pipeline export`. It's not implemented here, let me know if I need to explain more

File: agent/src/agent/cli/pipeline.py [hunk 2]
```diff
@@ -339,14 +340,31 @@ def export(dir_path):
     """
     Export pipelines' config into file
     """
+
+    def _export_file(path, pipeline_name):
```
Comment (dariakharlan) [inline agent/src/agent/cli/pipeline.py:344]: the file might not exist anymore inside the container (for example if the container is recreated) you need to copy the content of the file from the config that is stored in the dataabse

File: agent/src/agent/cli/pipeline.py [hunk 3]
```diff
@@ -339,14 +340,31 @@ def export(dir_path):
...
+        destination = os.path.join(os.path.dirname(dir_path), 'processing_files', pipeline_name)
```
Comment (dariakharlan) [inline agent/src/agent/cli/pipeline.py:345]: the file name should be the same as original value, destination path is the same dir

**Review by dariakharlan — CHANGES_REQUESTED** [review]

**Review by dariakharlan — APPROVED** [review]

**Pull request merged**